# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeChild.cpp
# Commit: 46c9b479034f
# Full Hash: 46c9b479034f2c6ee710d8d5a703de4a07e77557
# Author: Alexis Beingessner <a.beingessner@gmail.com>
# Date: 2019-07-12 21:43:58
# Regressor Bug: 1547351
# File Overlap Count: 2
# Description:
#   Bug 1547351 - Automatically set content process render root r=kats
#   
#   This implements the idea of automatically setting a content proc's
#   render root based on the render root enclosing the iframe that
#   points to it. There was a bit of cleanup in here that was a bit
# ==============================================================================

diff -r 4689b2870416 -r 46c9b479034f gfx/layers/wr/WebRenderBridgeChild.cpp
--- a/gfx/layers/wr/WebRenderBridgeChild.cpp	Fri Jul 12 14:14:22 2019 +0000
+++ b/gfx/layers/wr/WebRenderBridgeChild.cpp	Fri Jul 12 14:14:22 2019 +0000
@@ -122,6 +122,7 @@
                (XRE_IsParentProcess() &&
                 StaticPrefs::gfx_webrender_split_render_roots()));
     renderRoot.mCommands = std::move(mParentCommands[renderRoot.mRenderRoot]);
+    renderRoot.mIdNamespace = mIdNamespace;
   }
 
   nsTArray<CompositionPayload> payloads;
@@ -129,11 +130,10 @@
     mManager->TakeCompositionPayloads(payloads);
   }
 
-  this->SendSetDisplayList(std::move(aRenderRoots), mDestroyedActors,
-                           GetFwdTransactionId(), aTransactionId, mIdNamespace,
-                           aContainsSVGGroup, aVsyncId, aVsyncStartTime,
-                           aRefreshStartTime, aTxnStartTime, aTxnURL, fwdTime,
-                           payloads);
+  this->SendSetDisplayList(
+    std::move(aRenderRoots), mDestroyedActors, GetFwdTransactionId(),
+    aTransactionId, aContainsSVGGroup, aVsyncId, aVsyncStartTime,
+    aRefreshStartTime, aTxnStartTime, aTxnURL, fwdTime, payloads);
 
   // With multiple render roots, we may not have sent all of our
   // mParentCommands, so go ahead and go through our mParentCommands and ensure
@@ -146,8 +146,8 @@
 void WebRenderBridgeChild::EndEmptyTransaction(
     const FocusTarget& aFocusTarget,
     nsTArray<RenderRootUpdates>& aRenderRootUpdates,
-    uint32_t aPaintSequenceNumber, TransactionId aTransactionId,
-    const mozilla::VsyncId& aVsyncId, const mozilla::TimeStamp& aVsyncStartTime,
+    TransactionId aTransactionId, const mozilla::VsyncId& aVsyncId,
+    const mozilla::TimeStamp& aVsyncStartTime,
     const mozilla::TimeStamp& aRefreshStartTime,
     const mozilla::TimeStamp& aTxnStartTime, const nsCString& aTxnURL) {
   MOZ_ASSERT(!mDestroyed);
@@ -168,10 +168,9 @@
   }
 
   this->SendEmptyTransaction(
-      aFocusTarget, aPaintSequenceNumber, std::move(aRenderRootUpdates),
-      mDestroyedActors, GetFwdTransactionId(), aTransactionId, mIdNamespace,
-      aVsyncId, aVsyncStartTime, aRefreshStartTime, aTxnStartTime, aTxnURL,
-      fwdTime, payloads);
+      aFocusTarget, std::move(aRenderRootUpdates), mDestroyedActors,
+      GetFwdTransactionId(), aTransactionId, aVsyncId, aVsyncStartTime,
+      aRefreshStartTime, aTxnStartTime, aTxnURL, fwdTime, payloads);
 
   // With multiple render roots, we may not have sent all of our
   // mParentCommands, so go ahead and go through our mParentCommands and ensure